ludwig/automl/defaults/tabnet_config.yaml

```diff
@@ -0,0 +1,73 @@
+combiner:
+  type: tabnet
+  size: 32 # n_a
+  output_size: 32 # n_d
+  num_steps: 5
+  sparsity: 0.0001 # lambda_sparse
+  bn_virtual_bs: 512 # b_v
+  bn_momentum: 0.7 # m_b
+  relaxation_factor: 1.5 # gamma
+
+training:
+  # validation_metric: accuracy
+  batch_size: auto #2048
+  learning_rate: auto #0.02
+  decay: true
+  decay_steps: 500
+  decay_rate: 0.95
+  epochs: 3000
+  # early_stop: 300
+  regularization_lambda: 1
+  optimizer:
+    type: adam
+  tune_batch_size:
+    type: bin_search
+    substitute_with_max: True
+
+hyperopt:
+  # goal: maximize
+  parameters:
+    training.learning_rate:
```

Comment:
@tgaddair Reasoning behind the auto keyword is as follows. We want to tune the learning_rate and scale the batch_size only when those parameters are not part of the hyperparameter search space. The challenge with tuning the parameters during a hyperparameter optimization experiment is that the tuning algorithm has no knowledge that the value of the parameter it sample has changed and thus learns an incorrect parameter search space. This being said, I think the default behavior should be to do a hyperparameter search over learning_rate and batch_size (because we learn a relationship between batch_size and lr) vs tuning them independently. We can of course run experiments comparing the efficacy of tuning vs. search space optimization and change the default behavior.

Reply:
I see, so in the current implementation the `auto` keyword will be overridden by the values selected from the search space.

Reply:
yup!